### PR TITLE
ReceiptDomain: custom trace upper bound

### PR DESCRIPTION
**Pull request by AskAlexSharov**

### CODE REVIEW

File: cmd/state/exec3/historical_trace_worker.go [hunk 1]
```diff
@@ -240,122 +245,100 @@ type ExecArgs struct {
...
+	g, ctx := errgroup.WithContext(ctx)
 
 	// can afford big limits - because historical execution doesn't need conflicts-resolution
-	resultChannelLimit := workerCount * 128
-	heapLimit := workerCount * 128
-	rws := state.NewResultsQueue(resultChannelLimit, heapLimit) // workerCount * 4
-
-	reducerGroup := &errgroup.Group{}
+	resultChannelLimit := workerCount * 16
+	heapLimit := workerCount * 16
```
Comment (JkLondon) [inline cmd/state/exec3/historical_trace_worker.go:253]: why decreasing?

File: cmd/state/exec3/historical_trace_worker.go [hunk 2]
```diff
@@ -420,7 +405,7 @@ func CustomTraceMapReduce(fromBlock, toBlock uint64, consumer TraceConsumer, ctx
 	}
 	logEvery := time.NewTicker(1 * time.Second)
 	defer logEvery.Stop()
-	for blockNum := fromBlock; blockNum <= toBlock; blockNum++ {
+	for blockNum := fromBlock; blockNum < toBlock && !workersExited.Load(); blockNum++ {
```
Comment (JkLondon) [inline cmd/state/exec3/historical_trace_worker.go:408]: it's because [from, to) but we've decreased toBlock already.. would it be a problem?

Comment (AskAlexSharov) [reply cmd/state/exec3/historical_trace_worker.go]: don't know - such corner-cases can only be checked by unit-tests. changed to `<=`

File: cmd/state/exec3/historical_trace_worker.go (hunk 1 above)
Comment (AskAlexSharov) [reply cmd/state/exec3/historical_trace_worker.go]: reverted

**Review by JkLondon — APPROVED** [review]

File: erigon-lib/state/aggregator.go [hunk 3]
```diff
@@ -943,7 +943,7 @@ func (ac *AggregatorRoTx) PruneSmallBatchesDb(ctx context.Context, timeout time.
 
 	var pruneLimit uint64 = 1_000
 	if furiousPrune {
-		pruneLimit = 1_000_000
+		pruneLimit = 10_000_000
```
[the remark this answers is not in the document]
Comment (AskAlexSharov) [reply erigon-lib/state/aggregator.go:946]: yes. but let's abandon this change. seems not useful.
